op-batcher/batcher/driver.go

```diff
@@ -635,32 +664,31 @@ func (l *BatchSubmitter) waitNodeSync() error {
 }
 
 // publishStateToL1 queues up all pending TxData to be published to the L1, returning when there is no more data to
-// queue for publishing or if there was an error queing the data.  maxDuration tells this function to return from state
-// publishing after this amount of time has been exceeded even if there is more data remaining.
-func (l *BatchSubmitter) publishStateToL1(queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef], daGroup *errgroup.Group, maxDuration time.Duration) {
-	start := time.Now()
+// queue for publishing or if there was an error queing the data.
+func (l *BatchSubmitter) publishStateToL1(ctx context.Context, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef], daGroup *errgroup.Group) {
 	for {
-		// if the txmgr is closed, we stop the transaction sending
-		if l.Txmgr.IsClosed() {
-			l.Log.Info("Txmgr is closed, aborting state publishing")
-			return
-		}
-		if !l.checkTxpool(queue, receiptsCh) {
-			l.Log.Info("txpool state is not good, aborting state publishing")
+		select {
+		case <-ctx.Done():
 			return
-		}
+		default:
```

Comment:
You could move the code inside the `default` case just after the `select` to save one indentation.

Why does the function now receive a `ctx`, but passes the `l.killCtx` to the actual `l.publishTxToL1` call? In isolation, this looks confusing.

Reply:
0f27d9e4f36dec7f2ac3bf17de37160944f14f31 and 25785651080390fae96ee3d6d2581dca5ad11883

Reply:
I'm wondering now, why do we even pass the kill but not the shutdown context to the queue?

Reply:
From what I can tell, there are two shutdown modes for our op-services (including the batcher). Graceful shutdown (the first interrupt signal) or forced shutdown / kill (the second interrupt signal). 

On graceful shutdown, we wait for in flight transactions from the queue to confirm or reject. On forced shutdown, we do not. 

Our e2e tests appear to use a forced shutdown.

Happy to dig into this further, but I think it is somewhat outside scope or could at least be tackled in a followup.

Reply:
ok that makes sense, to wait for in-flight txs. Let's leave it where it is then.